### ISSUE TITLE
Custom module still showing after receving update signal even though `exec-if` returns false

### ISSUE THREAD

**Issue by whtv** [Alexays/Waybar#637]
It seems that when waybar is sent the configured `signal`, the custom module `exec` is run. If I understand correctly, I think `exec-if` should be called instead.

For example, my custom pacman module
```
"custom/pacman": {
        "format": "  {}",
        "interval": 86400,
        "exec": "checkupdates | wc -l",
        "exec-if": "checkupdates",
        "on-click": "termite --hold -e 'sudo pacman -Syu' ; pkill -SIGRTMIN+8 waybar",
        "signal": 8
}
```
only shows up if checkupdates returns any packages (otherwise checkupdates exits non-zero).
The way the signal is handled now, after performing the pacman upgrade (`on-click`) the module will sit there with a ‘0’ until it is due again, 24 hours in the worst case.

**Comment by whtv** [comment]
Hmm I got it wrong, sorry. Looking at `custom.cpp`, it _shoud_ actually check `exec-if` after receiving an update signal, so in my case, it should hide the module after performing a pacman upgrade; but it doesn’t. Anyway, there’s a workaround for this specific use case:
`"exec": "checkupdates | wc -l | grep -vx 0"` will exit non-zero if there’s 0 packages and hide the module.

**Comment by Alexays** [comment]
Are you able to debug this?
Check around this line why the return code is 0
https://github.com/Alexays/Waybar/blob/10b152ac3e28c59c2ec818c5b34351f620420e44/src/modules/custom.cpp#L29

**Comment by whtv** [comment]
I’m not sure how I woud go about debugging this. The return code should definitely be non-zero (2 actually), as is confirmed by the fact that restarting waybar immediately after the module is refreshed will properly hide the module. I also tried
`"exec-if": "[ $(checkupdates | wc -l) -gt 0 ]"` with the same result: after receiving the signal, the module is shown with the output `  0`. So somehow the `hide()` function does not work properly after the `delayWorker` thread is woken up.

**Comment by Alexays** [comment]
Should be fixed with 3b16946c25cf3088b29336720b7afa0551e7598f
Can you test on master?
